be/src/olap/base_compaction.cpp

```diff
@@ -86,6 +86,15 @@ Status BaseCompaction::execute_compact() {
 }
 
 void BaseCompaction::_filter_input_rowset() {
+    auto rs_iter = _input_rowsets.begin();
+    while (rs_iter != _input_rowsets.end()) {
+        if (_input_rowsets.size() > config::base_compaction_max_rowset_num) {
+            rs_iter = _input_rowsets.erase(rs_iter);
+        } else {
+            break;
+        }
+    }
+
```

Comment:
Should be a defer, because there are other places where rowsets are removed.

Reply:
done